### PR TITLE
Add scaladoc examples for Vec and Bundle

### PR DESCRIPTION
**Pull request by chick**
Added some simple example for Vec and Bundle creation.
Comments welcome

### CODE REVIEW

File: chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala [hunk 1]
```diff
@@ -160,6 +160,21 @@ object Vec {
   * the last one takes effect (unlike Mem, where the result is undefined)
   * @note Vecs, unlike classes in Scala's collection library, are propagated
   * intact to FIRRTL as a vector type, which may make debugging easier
+  *
+  * Careful consideration should be given over the use of [[Vec]] vs [[Seq]] or some other scala collection. In
+  * general [[Vec]] only needs to be used when there is a need to express the hardware collection in a [[Reg]]
+  * or IO [[Bundle]] or when access to elements of the array is indexed via a hardware signal.
+  * Example of indexing into a [[Vec]] using a hardware address and where the [[Vec]] is defined in an IO [[Bundle]]
+  *  {{{
```
Comment (ducky64) [inline chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala]: Consider use of the `@example` tag?

Comment (ducky64) [inline chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala]: Do you need the full Module, or can the example be more minimal?

File: chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala [hunk 2]
```diff
@@ -410,6 +425,34 @@ abstract class Record extends Aggregate {
   *
   * Usage: extend this class (either as an anonymous or named class) and define
   * members variables of [[Data]] subtypes to be elements in the Bundle.
+  *
+  * Example of an anonymous IO bundle
```
Comment (ducky64) [inline chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala]: `@example` tag?

**Comment by ducky64** [comment]
Did you miss tagging the second example? In Bundle?

**Comment by chick** [comment]
@ducky64 as a matter of fact I did miss second @example, tag fixed now

**Comment by jackkoenig** [comment]
How does it actually look with the example tags? I think it might move them away so that text in between them may not make sense

**Comment by ducky64** [comment]
You can preview by building the scaladoc?

**Comment by jackkoenig** [comment]
The `@example` tag doesn't seem to be doing anything. And the first example in Bundle still doesn't have it

**Comment by jackkoenig** [comment]
Actually, I was behind, the tag does move the example elsewhere, so maybe the tag isn't right for the style here? [Scala List](https://github.com/scala/scala/blob/v2.12.1/src/library/scala/collection/immutable/List.scala#L36) seems to use the tag sometimes but not always depending on what they're trying to do

**Comment by chick** [comment]
@jackkoenig That is weird stuff, looking into it

**Pull request merged**